src/clip.sol

```diff
@@ -227,7 +229,7 @@ contract Clipper {
                   uint256 max,          // Maximum acceptable price (DAI / collateral) [ray]
                   address who,          // Receiver of collateral, payer of DAI, and external call address
                   bytes calldata data   // Data to pass in external call; if length 0, no call is done
-    ) external lock isStopped(2) {
+    ) external note lock isStopped(2) {
```

Comment:
Comment regarding the custom Take event `emit Take(id, sale.tab, sale.lot, sale.usr);`

I recommend adding `owe` and `price` to improve the integrator's experience of determining participation size and price, respectively. Although one could reconstruct those values, with knowledge of previous tab / lots and reading the price curve, as defined in the `abacus`, I think it adds unnecessary friction for the shallow integration / analysis project. 

IIUC, `LogNote` does not record more than two function parameters, so I also think that `max` should be added, as it would be an interesting data point to consider when inspecting keeper behavior.

Reply:
Adding `owe`, `price` and `max`.